fe/fe-core/src/main/java/com/starrocks/sql/ast/SelectRelation.java

```diff
@@ -14,36 +16,43 @@
 import java.util.stream.Collectors;
 
 public class SelectRelation extends QueryRelation {
-    private final Expr predicate;
-
-    private final List<Expr> groupBy;
-    private final List<FunctionCallExpr> aggregate;
-    private final List<List<Expr>> groupingSetsList;
-    private final Expr having;
-    private final List<Expr> groupingFunctionCallExprs;
-
-    private final List<OrderByElement> sortClause;
-    private LimitElement limit;
-
-    private final boolean isDistinct;
-
-    private final List<AnalyticExpr> outputAnalytic;
-    private final List<AnalyticExpr> orderByAnalytic;
-
-    private final Scope orderScope;
-
+    /**
+     * selectList is created by parser
+     * and will be converted to outputExpr in Analyzer
+     */
+    private SelectList selectList;
     /**
      * out fields is different with output expr
      * output fields is represent externally visible resolved names
      * such as "k+1 as alias_name", k+1 is ArithmericExpr, alias_name is the output field
      */
     private List<Expr> outputExpr;
 
+    private final Expr predicate;
+
+    /**
+     * groupByClause is created by parser
+     * and will be converted to groupBy and groupingSetsList in Analyzer
+     */
+    private GroupByClause groupByClause;
+    private List<Expr> groupBy;
+    private List<FunctionCallExpr> aggregate;
+    private List<List<Expr>> groupingSetsList;
+    private Expr having;
+    private List<Expr> groupingFunctionCallExprs;
+
+    private boolean isDistinct;
+
+    private List<AnalyticExpr> outputAnalytic;
+    private List<AnalyticExpr> orderByAnalytic;
```

Comment:
Why AnalyticExpr need to distinguish output and orderby?

Reply:
Window functions can appear either in Output or in order by. Here is where the record appears. The transformer output needs to be processed separately. This is not a variable introduced by this PR, the diff appears here, just because the code location has changed